tools/devtool

```diff
@@ -302,6 +308,20 @@ validate_version() {
 
 }
 
+# Validate the user supplied kernel version number.
+# It must be composed of 2 groups of integers separated by dot, with an optional third group.
+validate_kernel_version() {
+    declare version_regex="^([0-9]+.)[0-9]+(.[0-9]+)?$"
```

Comment:
nit: We can use `local` to denote that the variable is used only within this scope. `declare` is not really necessary.